crates/nu-cli/src/cli.rs

```diff
@@ -157,26 +181,21 @@ pub fn load_plugins(context: &mut Context) -> Result<(), ShellError> {
                     }
                 };
 
+                // allow plugins with extensions on all platforms
                 let is_valid_name = {
-                    #[cfg(windows)]
-                    {
-                        bin_name
-                            .chars()
-                            .all(|c| c.is_ascii_alphanumeric() || c == '_' || c == '.')
-                    }
-
-                    #[cfg(not(windows))]
-                    {
-                        bin_name
-                            .chars()
-                            .all(|c| c.is_ascii_alphanumeric() || c == '_')
-                    }
+                    bin_name
+                        .chars()
+                        .all(|c| c.is_ascii_alphanumeric() || c == '_' || c == '.')
```

Comment:
Agree. It's better to look for certain file extensions rather than check if it's an executable (this is original behavior back then that slowed Nu startup down significantly)